src/main/java/org/traccar/handler/events/FuelEventHandler.java

```diff
@@ -50,18 +52,31 @@ public void onPosition(Position position, Callback callback) {
                 double before = lastPosition.getDouble(Position.KEY_FUEL_LEVEL);
                 double after = position.getDouble(Position.KEY_FUEL_LEVEL);
                 double change = after - before;
+                double difference = Math.abs(after - before);
 
                 if (change > 0) {
                     double threshold = AttributeUtil.lookup(
                             cacheManager, Keys.EVENT_FUEL_INCREASE_THRESHOLD, position.getDeviceId());
                     if (threshold > 0 && change >= threshold) {
-                        callback.eventDetected(new Event(Event.TYPE_DEVICE_FUEL_INCREASE, position));
+                        Map<String, Object> attributes = new HashMap<>();
+                        attributes.put("before", before);
+                        attributes.put("after", after);
+                        attributes.put("change", change);
```

Comment:
I don't think it makes sense to have all three parameters. I think either before/after or just the change would be enough.